metagpt/provider/bedrock/amazon_bedrock_api.py

```diff
@@ -0,0 +1,136 @@
+from typing import Literal
+import json
+from metagpt.const import USE_CONFIG_TIMEOUT
+from metagpt.provider.llm_provider_registry import register_provider
+from metagpt.configs.llm_config import LLMConfig, LLMType
+from metagpt.provider.base_llm import BaseLLM
+from metagpt.logs import log_llm_stream, logger
+from metagpt.provider.bedrock.bedrock_provider import get_provider
+from metagpt.provider.bedrock.utils import NOT_SUUPORT_STREAM_MODELS, get_max_tokens
+try:
+    import boto3
+    from botocore.eventstream import EventStream
+except ImportError:
+    raise ImportError(
+        "boto3 not found! please install it by `pip install boto3` ")
+
+
+@register_provider([LLMType.AMAZON_BEDROCK])
+class AmazonBedrockLLM(BaseLLM):
+    def __init__(self, config: LLMConfig):
+        self.config = config
+        self.__client = self.__init_client("bedrock-runtime")
+        self.__provider = get_provider(self.config.model)
+        logger.warning(
+            "Amazon bedrock doesn't support asynchronous now")
+
+    def __init_client(self, service_name: Literal["bedrock-runtime", "bedrock"]):
+        """initialize boto3 client"""
+        # access key and secret key from https://us-east-1.console.aws.amazon.com/iam
+        self.__credentital_kwargs = {
+            "aws_secret_access_key": self.config.secret_key,
+            "aws_access_key_id": self.config.access_key,
+            "region_name": self.config.region_name
+        }
+        session = boto3.Session(**self.__credentital_kwargs)
+        client = session.client(service_name)
+        return client
+
+    def _get_client(self):
```

Comment:
_get_client use
@property
def client(self):
    return self._client?

same with _get_provider. but it seems that _get_client doesn't used in anyplace?

Reply:
It is used in test